ff/serving/metadata/metadata_test.go

```diff
@@ -0,0 +1,1105 @@
+}
+
+func makeStreamMockLabel() *LabelStreamMock {
+	rec := make(chan *pb.Name, 10)
+	sent := make(chan *pb.Label, 10)
+	return &LabelStreamMock{
+		ctx:            context.Background(),
+		recvToServer:   rec,
+		sentFromServer: sent,
+	}
+}
+
+type LabelStreamMock struct {
+	grpc.ServerStream
+	ctx            context.Context
+	recvToServer   chan *pb.Name
+	sentFromServer chan *pb.Label
+}
+
+func (m *LabelStreamMock) Context() context.Context {
+	return m.ctx
+}
+func (m *LabelStreamMock) Send(resp *pb.Label) error {
+	m.sentFromServer <- resp
+	return nil
+}
+func (m *LabelStreamMock) Recv() (*pb.Name, error) {
+	req, more := <-m.recvToServer
+	if !more {
+		return nil, errors.New("no more to recieve")
+	}
+	return req, nil
+}
+
+func (m *LabelStreamMock) SendFromClient(req *pb.Name) error {
+	m.recvToServer <- req
+	return nil
+}
+func (m *LabelStreamMock) RecvToClient() (*pb.Label, error) {
+	response, more := <-m.sentFromServer
+	if !more {
+		return nil, errors.New("no more to recieve")
+	}
+	return response, nil
+}
+
+func makeStreamMockSource() *SourceStreamMock {
+	rec := make(chan *pb.Name, 10)
+	sent := make(chan *pb.Source, 10)
+	return &SourceStreamMock{
+		ctx:            context.Background(),
+		recvToServer:   rec,
+		sentFromServer: sent,
+	}
+}
+
+type SourceStreamMock struct {
+	grpc.ServerStream
+	ctx            context.Context
+	recvToServer   chan *pb.Name
+	sentFromServer chan *pb.Source
+}
+
+func (m *SourceStreamMock) Context() context.Context {
+	return m.ctx
+}
+func (m *SourceStreamMock) Send(resp *pb.Source) error {
+	m.sentFromServer <- resp
+	return nil
+}
+func (m *SourceStreamMock) Recv() (*pb.Name, error) {
+	req, more := <-m.recvToServer
+	if !more {
+		return nil, errors.New("no more to recieve")
+	}
+	return req, nil
+}
+
+func (m *SourceStreamMock) SendFromClient(req *pb.Name) error {
+	m.recvToServer <- req
+	return nil
+}
+func (m *SourceStreamMock) RecvToClient() (*pb.Source, error) {
+	response, more := <-m.sentFromServer
+	if !more {
+		return nil, errors.New("no more to recieve")
+	}
+	return response, nil
+}
+
+func TestCreateMetadataserverMethods(t *testing.T) {
+	sugar := zap.NewExample().Sugar()
+	server := Config{
+		Logger: sugar,
+	}
+	metadataServer, err := NewMetadataServer(&server)
+	if err != nil {
+		t.Fatalf("Failed to set up metadataserver %s", err)
+	}
+	if _, err := metadataServer.CreateUser(context.Background(), &pb.User{Name: "Simba Khadder"}); err != nil {
+		t.Fatalf("Failed to create user%s", err)
+	}
+	stream := makeStreamMock()
+	if err := stream.SendFromClient(&pb.Name{Name: "Simba Khadder"}); err != nil {
+		t.Fatalf("Failed to send from client user%s", err)
+	}
+	go func() {
+		if err := metadataServer.GetUsers(stream); err != nil {
+			t.Fatalf("Failed to get users%s", err)
+		}
+	}()
+	go func() {
+		var v pb.Empty
+		if err := metadataServer.ListUsers(&v, stream); err != nil {
+			t.Fatalf("Failed to list users%s", err)
+		}
+	}()
+	_, err2 := stream.RecvToClient()
```

Comment:
you should never have numbers like this in your code, this can be inlined with the line below this